cypress/integration/other/configuration.spec.js

```diff
@@ -126,4 +126,26 @@ describe('Configuration', () => {
       );
     });
   });
+
+  describe('suppressErrorRendering', () => {
+    it('should not render error diagram if suppressErrorRendering is set', () => {
+      const url = 'http://localhost:9000/suppressError.html?suppressErrorRendering=true';
+      cy.viewport(1440, 1024);
+      cy.visit(url);
+      cy.get('#test');
+      cy.matchImageSnapshot(
+        'configuration.spec-should-not-render-error-diagram-if-suppressErrorRendering-is-set'
+      );
+    });
+
+    it('should render error diagram if suppressErrorRendering is not set', () => {
+      const url = 'http://localhost:9000/suppressError.html';
+      cy.viewport(1440, 1024);
+      cy.visit(url);
+      cy.get('#test');
```

Comment:
I'm getting a race-condition when running the E2E test. The following seems to fix it:

```suggestion
      cy.get('#test').find("svg").should(($svg) => {
        expect($svg).to.have.length(5); // all five diagrams should be rendered!
        $svg.each((_index, svg) => {
          expect(cy.$$(svg)).to.be.visible;
        });

        // some of the diagrams should be error diagrams
        expect($svg).to.contain("Syntax error");
      });
```

### Screenshot of before and after

![configuration spec-should-render-error-diagram-if-suppressErrorRendering-is-not-set diff](https://github.com/mermaid-js/mermaid/assets/19716675/01215341-150d-4b5e-a60e-874fa2c84ecb)

Reply:
Added the checks.

Reply:
Looks good! If you really want to, you could also replace the current Cypress code by using the `urlSnapshotTest` function in https://github.com/mermaid-js/mermaid/blob/4201e4775d9206a5fc44006a2e26747cfc10ef78/cypress/helpers/util.ts#L75-L82, which checks for `.rendered`, but :shrug:, what we have already works, so there's no need to change anything.

I'll let Knut officially approve this PR, just in case there's some security issue with `secure` keys that I haven't spotted.